source/js/components/petition/petition.jsx

```diff
@@ -528,7 +513,7 @@ export default class Petition extends React.Component {
     });
 
     let emailGroupClass = classNames({
-      'has-danger': this.state.userTriedSubmitting && (!this.email.element.value || !this.validatesAsEmail(this.email.element.value))
```

Comment:
As we are removing the function, but, we still want the email to not be null, hence I left the checking of the element's value.